core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/SenderThreadLocalBackOffData.java

```diff
@@ -92,34 +92,69 @@ public void onDoneSending() {
      *         4. The thread has exhausted all the back-off timeouts
      */
     public boolean backOff() {
-         try {
-             lock.lock();
-             ++currentBackOffIndex;
-             if (currentBackOffIndex == backOffTimeoutsInMillis.length) {
-                 currentBackOffIndex = -1;
-
-                 // Exhausted the back-offs
-                 return false;
-             }
+        try {
+            lock.lock();
+            ++currentBackOffIndex;
+            if (currentBackOffIndex == backOffTimeoutsInMillis.length) {
+                currentBackOffIndex = -1;
 
-             if (!instanceIsActive) {
+                // Exhausted the back-offs
                 return false;
             }
 
-             try {
-                 long millisecondsToWait = backOffTimeoutsInMillis[currentBackOffIndex];
-                 if (millisecondsToWait > BackOffTimesPolicy.MIN_TIME_TO_BACK_OFF_IN_MILLS) {
-                     millisecondsToWait += addMilliseconds;
-                 }
-                 backOffCondition.await(millisecondsToWait, TimeUnit.MILLISECONDS);
-                 return instanceIsActive;
-            } catch (InterruptedException e) {
-                return false;
+            if (!instanceIsActive) {
```

Comment:
@debugthings is it true that thread will never starve in the same state (i.e blocked state). If not we might always hit this condition for certain thread.

Reply:
So, this is an artifact of the blocking backoff implementation (which I left intact). The only way for this to get set is if someone calls stop(). It's highly unlikely that it will be called since it's not referenced anywhere in the code.

It's in place to make sure if someone wants to disable the backoff they can do so with stop. Not surprisingly there is no start method so once the manager is stopped that's it

Reply:
If there is a feature like this to disable the backoff it should be documented. And if there is no way to start if back then stop also doesn't make sense to me. Either case I think if we want to keep it just comment on it , regarding it is not advised to use this at any point or just remove the method if we do not plan to have this functionality.

Reply:
I am all for deleting code that is not used.

Reply:
I left this code intact as there are some old tests that are dependent on it and that is too much to tackle in this PR. Created task #559 to follow up